src/test_utils.rs

```diff
@@ -18,44 +18,59 @@ pub(crate) struct TestContext {
     pub dir: TempDir,
 }
 
-/// Create a new, opened [TestContext] using given callback.
-///
-/// The [Context] will be opened with the SQLite database named
-/// "db.sqlite" in the [TestContext.dir] directory.
-///
-/// [Context]: crate::context::Context
-pub(crate) async fn test_context() -> TestContext {
-    let dir = tempdir().unwrap();
-    let dbfile = dir.path().join("db.sqlite");
-    let ctx = Context::new("FakeOs".into(), dbfile.into()).await.unwrap();
-    TestContext { ctx, dir }
-}
+impl TestContext {
+    /// Create a new [TestContext].
+    ///
+    /// The [Context] will be created and have an SQLite database anemd "db.sqlite" in the
+    /// [TestContext.dir] directory.  This directory is cleaned up when the [TestContext] is
+    /// dropped.
+    ///
+    /// [Context]: crate::context::Context
+    pub async fn new() -> Self {
+        let dir = tempdir().unwrap();
+        let dbfile = dir.path().join("db.sqlite");
+        let ctx = Context::new("FakeOS".into(), dbfile.into()).await.unwrap();
+        Self { ctx, dir }
+    }
 
-/// Return a dummy [TestContext].
-///
-/// The context will be opened and use the SQLite database as
-/// specified in [test_context] but there is no callback hooked up,
-/// i.e. [Context::call_cb] will always return `0`.
-pub(crate) async fn dummy_context() -> TestContext {
-    test_context().await
-}
+    /// Create a new configured [TestContext].
+    ///
+    /// This is a shortcut which automatically calls [TestContext::configure] after creating
+    /// the context.
+    pub async fn new_configured() -> Self {
+        let t = Self::new().await;
+        t.configure_alice().await;
```

Comment:
`configure_alice` used by `new_configured` also configures the key, while `configured_offline_context` function didn't.

This way two `new_configured` will use the same key, is this intended? Maybe rename `new_configured` into `new_alice`?

Reply:
> `configure_alice` used by `new_configured` also configures the key, while `configured_offline_context` function didn't.

Yes, there were two variations of "configuring" accounts with only small difference that I decided to merge for simplicity.  There's only a tiny performance hit for this.

> This way two `new_configured` will use the same key, is this intended? Maybe rename `new_configured` into `new_alice`?

It is intended, but the name suggestion is great, makes it clearer.